LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -30,96 +36,103 @@ internal sealed class ConcentratorDeduplication<T> :
 
         public ConcentratorDeduplication(
             IMemoryCache cache,
-            WebSocketWriterRegistry<StationEui, string> socketRegistry,
-            ILogger<IConcentratorDeduplication<T>> logger)
+            IDeduplicationStrategyFactory deduplicationStrategy,
+            ILogger<IConcentratorDeduplication> logger)
         {
             this.cache = cache;
-            this.socketRegistry = socketRegistry;
+            this.deduplicationStrategy = deduplicationStrategy;
             this.logger = logger;
         }
 
-        public bool ShouldDrop(T frame, StationEui stationEui)
+        public Result CheckDuplicate(LoRaRequest loRaRequest, LoRaDevice? loRaDevice)
         {
-            if (frame == null) throw new ArgumentNullException(nameof(frame));
+            _ = loRaRequest ?? throw new ArgumentNullException(nameof(loRaRequest));
 
-            var key = CreateCacheKey(frame);
+            var key = CreateCacheKey(loRaRequest);
+            var stationEui = loRaRequest.StationEui;
 
             StationEui previousStation;
             lock (this.cache)
             {
                 if (!this.cache.TryGetValue(key, out previousStation))
                 {
-                    AddToCache(key, stationEui);
-                    return false;
+                    _ = this.cache.Set(key, stationEui, new MemoryCacheEntryOptions()
+                    {
+                        SlidingExpiration = DefaultExpiration
+                    });
+                    return Result.NotDuplicate;
                 }
             }
 
-            if (previousStation == stationEui)
+            if (LoRaRequest.RequiresConfirmation(loRaRequest) && previousStation == stationEui)
             {
-                // considered as a resubmit
-                this.logger.LogDebug("Message received from the same EUI {StationEui} as before, will not drop.", stationEui);
-                return false;
+                this.logger.LogDebug($"Message received from the same EUI {stationEui} as before, will not drop.");
+                return Result.DuplicateDueToResubmission;
             }
 
             // received from a different station
-            if (this.socketRegistry.IsSocketWriterOpen(previousStation))
+            if (ShouldDrop(loRaRequest, loRaDevice))
             {
                 this.logger.LogInformation($"{Constants.DuplicateMessageFromAnotherStationMsg} with EUI {stationEui}, will drop.");
```

Comment:
information logs should be made sparse. This is a normal case that is potentially being spit out several times for every message (if we have drop and multiple concentrators connected). Can we change this to Debug?